ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -831,9 +832,58 @@ prop_TxSeq_lookupByTicketNo_sound smalls small =
     txseq =
         foldl' (TxSeq.:>) TxSeq.Empty $ map mkTicket haystack
 
-    mkTicket x = TxTicket x (mkTicketNo x)
+    mkTicket x = TxTicket x (mkTicketNo x) 0
     mkTicketNo = TicketNo . toEnum
 
+-- | Test that the 'fst' of the result of 'splitAfterTxSize' only contains
+-- 'TxTicket's whose summed up transaction sizes are less than or equal to
+-- that of the 'TxSizeInBytes' which the 'TxSeq' was split on.
+prop_TxSeq_splitAfterTxSize :: [Positive TxSizeInBytes] -> Property
+prop_TxSeq_splitAfterTxSize xs =
+    conjoin [ splitAndCheck txsz
+            | TxTicket _ _ txsz <- txTickets txseq ]
+  where
+    txSizes :: [TxSizeInBytes]
+    txSizes = map getPositive xs
+
+    txseq :: TxSeq Int
+    txseq = TxSeq.toTxSeq $ zip3 (repeat 0) (repeat (TicketNo 0)) txSizes
+
+    splitAndCheck :: TxSizeInBytes -> Bool
+    splitAndCheck txsz =
+      let (before, _after) = splitAfterTxSize txseq txsz
+      in  checkTxSizeSumLessThanOrEqual (txTickets before) txsz
+
+    txSizeSum :: [TxTicket tx] -> TxSizeInBytes
+    txSizeSum = sum . map txTicketTxSizeInBytes
+
+    checkTxSizeSumLessThanOrEqual :: [TxTicket tx] -> TxSizeInBytes -> Bool
+    checkTxSizeSumLessThanOrEqual txs txsz = txSizeSum txs <= txsz
+
+-- | Test that the results of 'splitAfterTxSizeSpec', a specification of
+-- 'splitAfterTxSize', match those of the real 'splitAfterTxSize'
+-- implementation.
+prop_TxSeq_splitAfterTxSizeSpec :: [Small TxSizeInBytes] -> Property
```

Comment:
Why `Small` instead of `Positive`?

Reply:
When I used `[Positive TxSizeInBytes]` I would run into a few counterexamples in which overflows were occurring when summing them up.

Reply:
Then I would use `choose (1, x)` where `x` is some reasonable value. See my next comment about `TxSizesAndMaxSize`.